hs/Database/CourseInsertion.hs

```diff
@@ -11,16 +11,40 @@ into the database. These functions are used as helpers for the WebParsing module
 module Database.CourseInsertion
     (insertCourse,
      setTutorialEnrolment,
-     setPracticalEnrolment) where
+     setPracticalEnrolment,
+     saveGraphJSON,
+     insertGraph) where
 
 import qualified Data.Text as T
-
+import qualified Data.ByteString.Lazy.Char8 as BSL
+import Happstack.Server.SimpleHTTP
 import Control.Monad.IO.Class (MonadIO)
 import Control.Monad.Trans.Reader (ReaderT)
 import Data.Maybe (fromMaybe)
-
-import Database.Persist.Sqlite (insert_, SqlBackend, (=.), (==.), updateWhere)
+import Config (databasePath)
+import Util.Happstack (createJSONResponse)
+import Database.Persist.Sqlite (fromSqlKey, update, insertMany_, insert_, insert, SqlBackend, (=.), (==.), updateWhere, runSqlite)
 import Database.Tables
+import Data.Aeson
+
+-- | Insert SVG graph and return its id
+insertGraph :: IO Response
+insertGraph = runSqlite databasePath $ do
+    gId <- insert $ Graph "" 256 256
+    update gId [GraphTitle =. ((++) "User generated graph " $ show $ fromSqlKey gId)]
```

Comment:
A new graph is inserted, and persistent generates a new `GraphId` for it, which is used to update the graph's title.